src/main/kotlin/com/example/ajaxproject/service/ChatServiceImpl.kt

```diff
@@ -0,0 +1,78 @@
+package com.example.ajaxproject.service
+
+import com.example.ajaxproject.dto.RoomDTO
+import com.example.ajaxproject.dto.SendMessageDTO
+import com.example.ajaxproject.exeption.NoSuchElementException
+import com.example.ajaxproject.exeption.UserNotFoundException
+import com.example.ajaxproject.model.ChatMessage
+import com.example.ajaxproject.model.ChatRoom
+import com.example.ajaxproject.repository.ChatMessageRepository
+import com.example.ajaxproject.repository.ChatRoomRepository
+import com.example.ajaxproject.repository.UserRepository
+import com.example.ajaxproject.service.interfaces.ChatService
+import org.bson.types.ObjectId
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import java.time.LocalDateTime
+
+@Service
+class ChatServiceImpl @Autowired constructor(
+    private val chatRoomRepository: ChatRoomRepository,
+    private val chatMessageRepository: ChatMessageRepository,
+    private val userRepository: UserRepository,
+) : ChatService {
+
+    override fun createRoom(senderId: Long, recipientId: Long): ChatRoom {
+
+        //if I had used authentication, this check would not have happened
+        userRepository.findById(senderId).orElseThrow { UserNotFoundException("Sender not found") }
+
+        userRepository.findById(recipientId).orElseThrow { UserNotFoundException("Recipient not found") }
+
+        val sortedUserIds = listOf(senderId, recipientId).sorted()
+
+        val roomId = "${sortedUserIds[0]}-${sortedUserIds[1]}"
```

Comment:
It's better to extract a function:

```kotlin
private fun buildRoomId(senderId: Long, recipientId: Long): String {
    // ....
    return "${sortedUserIds[0]}-${sortedUserIds[1]}"
}
```

Note: the function could be reused in `#getAllMessages`

----

It's not clear why we should do this
```kotlin
val sortedUserIds = listOf(senderId, recipientId).sorted()
```
Could you explain?

Reply:
I try to set id for Room as id of both users. But room 1-2 is the same as 2-1 . So thats why i wont to use smallerID-higherId in db.  But I agree that it can be simplified without a list. See in my new impl